mypy/modulefinder.py

```diff
@@ -751,12 +751,19 @@ def get_search_dirs(python_executable: str | None) -> tuple[list[str], list[str]
     else:
         # Use subprocess to get the package directory of given Python
         # executable
+        env = {**os.environ.__dict__, "PYTHONSAFEPATH": "1"}
```

Comment:
Not sure if this will work, but seems worth a shot:

```suggestion
        env = {**dict(os.environ), "PYTHONSAFEPATH": "1"}
```

Reply:
I know, but that's what was tried in https://github.com/python/mypy/pull/15122/commits/66db45d224f93f768811efa6e1627c25900cf763, and mypyc hated it for some reason: